src/colmap/geometry/gps.cc

```diff
@@ -197,4 +197,15 @@ std::vector<Eigen::Vector3d> GPSTransform::ENUToXYZ(
   return xyz;
 }
 
+std::ostream& operator<<(std::ostream& stream, const PosePrior& prior) {
+  const static Eigen::IOFormat kVecFmt(
+      Eigen::StreamPrecision, Eigen::DontAlignCols, ", ", ", ");
+  stream << "PosePrior(position=[" << prior.position.format(kVecFmt)
+         << "], position_covariance=["
+         << prior.position_covariance.format(kVecFmt)
+         << "], coordinate_system=" << static_cast<int>(prior.coordinate_system)
```

Comment:
The Python implementation prints the enum as string, is this not possible in C++?

Reply:
Needs a custom switch/case where we convert from enum to string.

Reply:
I have a custom macro for enum to string in C++. Will open a PR